src/python_inspector/resolution.py

```diff
@@ -116,6 +150,37 @@ def fetch_and_extract_sdist(repos, candidate, python_version):
     return os.path.join(utils_pypi.CACHE_THIRDPARTY_DIR, "extracted_sdists", sdist_file, sdist_file)
 
 
+def get_requirements_from_dependencies(dependencies, resolved_requirements):
+    """
+    Generate parsed requirements for the given ``dependencies``.
+    """
+    for dep in dependencies:
+        if not dep.purl:
+            continue
+
+        if dep.scope != "install":
+            continue
+
+        dep_purl = PackageURL.from_string(dep.purl)
+
+        dep_purl_name = packaging.utils.canonicalize_name(dep_purl.name)
+
+        if is_dep_resolved_and_in_resolved_requirements(
```

Comment:
@pombredanne this is what I have done to deal with pinned requirements. Please let me know what should be an ideal way for this. I use the pinned version of a requirement (that was seen first while doing the resolution) as a version for that requirement that will be seen further. The intuition behind this was that if we have a pinned version of any requirement then we don't need to resolve it further and we should provide the pinned version in our resolution and if we have 2 pinned versions of a req then in that case, the resolution will become impossible so I am using the first seen version, so we can provide a resolution.

Reply:
https://github.com/nexB/python-inspector/issues/46